src/function/arithmetic/round.js

```diff
@@ -147,23 +147,18 @@ export const createRound = /* #__PURE__ */ factory(name, dependencies, ({ typed,
       return x.round(n.toNumber())
     },
 
-    'Unit, number, Unit': typed.referToSelf(self => function (x, n, unit) {
+    'Unit, number | BigNumber, Unit': typed.referToSelf(self => function (x, n, unit) {
+      if (typeof n !== 'number') n = n.toNumber()
       const valueless = x.toNumeric(unit)
       return unit.multiply(self(valueless, n))
     }),
 
-    'Unit, BigNumber, Unit': typed.referToSelf(self => (x, n, unit) => self(x, n.toNumber(), unit)),
```

Comment:
It converted it to a number previously.  If there was BigNumber

Reply:
When removing the code `if (typeof n !== 'number') n = n.toNumber()` from the functions, all works except that the unit test that you refer to fails. 

The reason for this is the following: in general, when you use mixed numbers and BigNumbers in mathjs functions, the output will become a BigNumber (since a `BigNumber` can hold any `number` so that is a safe conversion, but a `number` cannot hold any `BigNumber`). The failing unit tests has a mix of `number` and `BigNumber` in its arguments. So it is fine when the output contains a BigNumber then: `unit(bignumber('3.12'), 'cm')` instead of `unit('3.12 cm')`. In the case of these rounding functions it may feel a bit odd, since the number of digits `n` is not the "main" value. This is an edge case though and if people really provide mixed `number` and `BigNumber` this way, `BigNumber` output is what they'll get ;). Does that sort of make sense?

In short: can you remove the lines `if (typeof n !== 'number') n = n.toNumber()` and adjust the concerning unit tests to expect a Unit with BigNumber value as output?

Reply:
Hmm. You are right though about that this behavior is already there in function `round`, so changing it would be a breaking change 🤔 . That is a pity. 

Thinking aloud...

Let's not introduce a breaking change here, and let's keep the behavior of function `round` as it is, and align the other functions with that (like you already have done right now).

So then there are two options I think:

1. have unified signatures like `Unit, number | BigNumber, Unit` and special logic `if (typeof n !== 'number') n = n.toNumber()`.
2. have separate signatures like `Unit, number, Unit` and `Unit, BigNumber, Unit`.

I think from these two options I like 2 best (your original approach): I prefer not having `typeof` checks in the logic and let all of that be handled via the function signatures. What do you think?


Sorry about the confusion and all the back-and-forth, I hand't realized that much of the behavior is already in function `round` 😕.

Reply:
Will go with option 2.

Reply:
Thanks, looks good. Nice to at least have 4 signatures instead of originally 6.